docs/decisions/0045-updated-vector-store-design.md

```diff
@@ -0,0 +1,860 @@
+    Task<bool> CollectionExistsAsync(string name, CancellationToken cancellationToken = default);
+    Task DeleteCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+// DB Specific Update implementations
+class AzureAISearchMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+class RedisMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+
+// Combined Create + Update Interface
+interface IMemoryCollectionService: IMemoryCollectionCreateService, IMemoryCollectionUpdateService {}
+
+// Base abstract class that forwards non-create operations to provided service.
+abstract class MemoryCollectionService(IMemoryCollectionUpdateService collectionsUpdateService): IMemoryCollectionService
+{
+    public abstract Task CreateCollectionAsync(string name, CancellationToken cancellationToken = default);
+    public IAsyncEnumerable<string> ListCollectionNamesAsync(CancellationToken cancellationToken = default) { return collectionsUpdateService.ListCollectionNamesAsync(cancellationToken); }
+    public Task<bool> CollectionExistsAsync(string name, CancellationToken cancellationToken = default) { return collectionsUpdateService.CollectionExistsAsync(name, cancellationToken); }
+    public Task DeleteCollectionAsync(string name, CancellationToken cancellationToken = default) { return collectionsUpdateService.DeleteCollectionAsync(name, cancellationToken); }
+}
+
+// Collections service implementations, that inherit from base class, and just adds the different creation implementations.
+class AzureAISearchChatHistoryMemoryService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+class AzureAISearchSemanticCacheMemoryService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+class AzureAISearchMLIndexMemoryService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+
+// Customer collections service implementation, that uses the base Azure AI Search implementation for get, doesExist and delete, but adds it's own creation.
+class ContosoProductsMemoryCollectionService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+
+```
+
+#### Option 5 - Separated collection and record management with collection create separate from other operations, with overall aggregation class on top.
+
+Same as option 3 / 4, plus:
+
+```cs
+
+interface IMemoryService : IMemoryCollectionCreateService, IMemoryCollectionService, IMemoryRecordService
+{    
+}
+
+// Create a static factory that produces one of these, so only the interface is public, not the class.
+internal class CombinedMemoryService<TDataModel>(IMemoryCollectionCreateService creation, IMemoryCollectionService collections, IMemoryRecordService<TDataModel> records): IMemoryService
+{
+}
+
+```
+
+#### Decision Outcome
```

Comment:
I don't see any example consumption code in this design. What does it look like for typical scenarios to query for the appropriate interfaces from DI and use them? For example, the above shows multiple implementations of MemoryCollectionService... how are those being registered in DI, how are those being queried for, etc.?

Reply:
I haven't designed the DI experience yet. I was hoping to focus on the core data interfaces first. That said, the scenario you mention is very important to get right. Let's chat about it directly, I'd like to get your input on what you'd like to see here.

Reply:
As discussed in person, I'll work on this in the next iteration.